pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1007,12 +1007,17 @@ public static JdbcCallParseInfo modifyJdbcCall(String jdbcSql, boolean stdString
           PSQLState.STATEMENT_NOT_ALLOWED_IN_FUNCTION_CALL);
     }
 
+    String suffix = " as result";
     if (serverVersion < 80100 /* 8.1 */ || protocolVersion != 3) {
-      sql = "select " + jdbcSql.substring(startIndex, endIndex) + " as result";
+      sql = new StringBuilder().append("select ").append(jdbcSql, startIndex, endIndex).append(suffix).toString();
```

Comment:
Is it worthwhile to make minor optimizations (with small negative impact on readability) for versions prior to 8.1?

Reply:
Up to you !
is this driver officially supports version prior to 8.2 ?

Reply:
Only protocol=3 is supported for now.
There was protocol=2 (for <8.2), however the implementation has already been removed.

Reply:
if not I'll remove the branch and the associated test